src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -62,7 +62,12 @@ protected override void Setup(FeatureConfigurationContext context)
             Func<string, IChannelReceiver> channelReceiverFactory;
             RegisterChannels(context, channelManager, out channelSenderFactory, out channelReceiverFactory);
 
-            gatewayPipeline.Register("GatewaySendProcessor", b => new GatewaySendBehavior(gatewayInputAddress, channelManager, new MessageNotifier(), b.Build<IDispatchMessages>(), context.Settings, CreateForwarder(channelSenderFactory, b.BuildAll<IDataBus>()?.FirstOrDefault()), CreateSiteRouters(context)), "Processes messages to be sent to the gateway");
+            gatewayPipeline.Register("ForwardFailedGatewayMessagesToErrorQueue", b => new ForwardFailedGatewayMessagesToErrorQueueBehavior(gatewayInputAddress, b.Build<CriticalError>()), "Handles failures when sending messages through the gateway");
+
+            var retryPolicy = context.Settings.HasSetting("Gateway.Retries.RetryPolicy") ? context.Settings.Get<Func<IncomingMessage, Exception, int ,TimeSpan>>("Gateway.Retries.RetryPolicy") : DefaultRetryPolicy.BuildWithDefaults();
+
+            gatewayPipeline.Register(new RetryFailedGatewayMessagesBehavior.Registration(gatewayInputAddress, retryPolicy, "ForwardFailedGatewayMessagesToErrorQueue"));
```

Comment:
passing in the dependency as a parameter but duplicating the string instead of using a variable seems weird.
Why not provide a constant for `"ForwardFailedGatewayMessagesToErrorQueue"` and reference that directly without passing it as a parameter?